src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -374,10 +379,16 @@ ParticipantConfiguration::getParticipants() const
 
 partition::ReceivedPartition::GeometricFilter ParticipantConfiguration::getGeoFilter(const std::string &geoFilter) const
 {
-  if (geoFilter == VALUE_FILTER_ON_MASTER) {
-    return partition::ReceivedPartition::GeometricFilter::ON_MASTER;
-  } else if (geoFilter == VALUE_FILTER_ON_SLAVES) {
-    return partition::ReceivedPartition::GeometricFilter::ON_SLAVES;
+  if (geoFilter == VALUE_FILTER_ON_MASTER || geoFilter == VALUE_FILTER_ON_PRIMARY) {
+    if (geoFilter == VALUE_FILTER_ON_MASTER) {
+      PRECICE_WARN("<on-master> tag is deprecated and will be removed in v3.0.0! Please use <on-primary> tag instead");
```

Comment:
```suggestion
      PRECICE_WARN("<on-master> tag is deprecated and will be removed in v3.0.0. Please use <on-primary> tag instead");
```
We decided to not use exclamation marks in error messages.